samples/python/reactors/surf_pfr.py

```diff
@@ -30,91 +32,46 @@
 
 t = tc + 273.15  # convert to Kelvin
 
-# import the gas model and set the initial conditions
-gas = ct.Solution(yaml_file, 'gas')
-gas.TPX = t, ct.one_atm, 'CH4:1, O2:1.5, AR:0.1'
-
-# import the surface model
-surf = ct.Interface(yaml_file, 'Pt_surf', [gas])
+# import the model and set the initial conditions
+surf = ct.Interface(yaml_file, 'Pt_surf')
 surf.TP = t, ct.one_atm
+gas = surf.adjacent['gas']
+gas.TPX = t, ct.one_atm, 'CH4:1, O2:1.5, AR:0.1'
 
-rlen = length/(NReactors-1)
-rvol = area * rlen * porosity
-
-# catalyst area in one reactor
-cat_area = cat_area_per_vol * rvol
-
-mass_flow_rate = velocity * gas.density * area
-
-# The plug flow reactor is represented by a linear chain of zero-dimensional
-# reactors. The gas at the inlet to the first one has the specified inlet
-# composition, and for all others the inlet composition is fixed at the
-# composition of the reactor immediately upstream. Since in a PFR model there
-# is no diffusion, the upstream reactors are not affected by any downstream
-# reactors, and therefore the problem may be solved by simply marching from
-# the first to last reactor, integrating each one to steady state.
-
-TDY = gas.TDY
-cov = surf.coverages
-
-print('    distance       X_CH4        X_H2        X_CO')
+mass_flow_rate = velocity * gas.density * area * porosity
 
 # create a new reactor
-gas.TDY = TDY
-r = ct.IdealGasReactor(gas, energy='off')
-r.volume = rvol
-
-# create a reservoir to represent the reactor immediately upstream. Note
-# that the gas object is set already to the state of the upstream reactor
-upstream = ct.Reservoir(gas, name='upstream')
-
-# create a reservoir for the reactor to exhaust into. The composition of
-# this reservoir is irrelevant.
-downstream = ct.Reservoir(gas, name='downstream')
-
-# Add the reacting surface to the reactor. The area is set to the desired
-# catalyst area in the reactor.
-rsurf = ct.ReactorSurface(surf, r, A=cat_area)
+r = ct.FlowReactor(gas)
+r.area = area
+r.surface_area_to_volume_ratio = cat_area_per_vol * porosity
+r.mass_flow_rate = mass_flow_rate
+r.energy_enabled = False
 
-# The mass flow rate into the reactor will be fixed by using a
-# MassFlowController object.
-m = ct.MassFlowController(upstream, r, mdot=mass_flow_rate)
-
-# We need an outlet to the downstream reservoir. This will determine the
-# pressure in the reactor. The value of K will only affect the transient
-# pressure difference.
-v = ct.PressureController(r, downstream, master=m, K=1e-5)
+# Add the reacting surface to the reactor
+rsurf = ct.ReactorSurface(surf, r)
 
 sim = ct.ReactorNet([r])
-sim.max_err_test_fails = 12
-
-# set relative and absolute tolerances on the simulation
-sim.rtol = 1.0e-9
-sim.atol = 1.0e-21
 
 output_data = []
+n = 0
+print('    distance       X_CH4        X_H2        X_CO')
+print('  {:10f}  {:10f}  {:10f}  {:10f}'.format(
+      0, *r.thermo['CH4', 'H2', 'CO'].X))
+
+while sim.time < length:
+    dist = sim.time * 1e3
```

Comment:
I know that distance is proportional to time, but it's still a little confusing to write it like this.

Reply:
I made some changes so that `ReactorNet` now has both `time` and `distance` properties, where the one that is not applicable (depending on the reactor types involved) raises an exception. I updated some of the documentation as well to reflect the fact that the integrator's independent variable could be either time or distance, but stopped short of creating alternative names for `set_initial_time()` and the `max_time_step` property.